### ISSUE TITLE
ScalaJSModule seems incompatible with ScoverageModule and ScoverageTests

### ISSUE THREAD

**Issue by carlosedp** [com-lihaoyi/mill#2267]
I have a ScalaJS module that I tried to add SCoverage thru the contrib plugin:

```scala
object frontend extends ScalaJSModule with Common with ScoverageModule {
  def scalaVersion     = versions.scala3
  def scalaJSVersion   = versions.scalajs
  def scoverageVersion = "2.0.7"
  def ivyDeps = super.ivyDeps() ++ Agg(
    ivy"org.scala-js::scalajs-dom::${versions.scalajsdom}",
  )

  def scalaJSUseMainModuleInitializer = true

  object test extends Tests with Common with TestModule.ScalaTest with ScoverageTests {
    // Test dependencies
    def ivyDeps = Agg(
      ivy"org.scalatest::scalatest::${versions.scalatest}",
    )
    def jsEnvConfig = T(JsEnvConfig.JsDom())
  }
}
```

I get the following error:

```
/Users/cdepaula/repos/scala-playground/zio-scalajs-stack/build.sc:148: object frontend inherits conflicting members:
  def toolsClasspath: mill.define.Target[mill.api.Loose.Agg[mill.api.PathRef]] (defined in trait ScalaJSModule) and
  def toolsClasspath: mill.T[mill.Agg[mill.api.PathRef]] (defined in trait ScoverageModule)
  (note: this can be resolved by declaring an `override` in object frontend.)
object frontend extends ScalaJSModule with Common with ScoverageModule {
       ^
/Users/cdepaula/repos/scala-playground/zio-scalajs-stack/build.sc:165: incompatible type in overriding
override def moduleDeps: Seq[mill.scalajslib.ScalaJSModule] (defined in trait Tests)
  with override def moduleDeps: Seq[mill.scalalib.JavaModule] (defined in trait ScoverageTests);
 found   : Seq[mill.scalalib.JavaModule]
 required: Seq[mill.scalajslib.ScalaJSModule]
  object test extends Tests with Common with TestModule.ScalaTest with ScoverageTests {
         ^
Compilation Failed
```

**Comment by lefou** [comment]
Can you please provide a reproducible example?

**Comment by lefou** [comment]
I tried with the following `build.sc`

```scala
import $ivy.`com.lihaoyi::mill-contrib-scoverage::`
import mill._
import scalalib._
import scalajslib._
import scalajslib.api._
import contrib.scoverage._

trait Common {
  object versions {
    val scala3 = "3.2.0"
    val scalajs = "1.12.0"
    val scalajsdom = "1.12.0"
    val scalatest = "3.2.15"
  }
}

object frontend extends ScalaJSModule with Common with ScoverageModule {
  def scalaVersion     = versions.scala3
  def scalaJSVersion   = versions.scalajs
  def scoverageVersion = "2.0.7"
  def ivyDeps = super.ivyDeps() ++ Agg(
    ivy"org.scala-js::scalajs-dom::${versions.scalajsdom}",
  )

  def scalaJSUseMainModuleInitializer = true

  object test extends Tests with Common with TestModule.ScalaTest with ScoverageTests {
    // Test dependencies
    def ivyDeps = Agg(
      ivy"org.scalatest::scalatest::${versions.scalatest}",
    )
    def jsEnvConfig = T(JsEnvConfig.JsDom())
  }
}

```

The first issue is already fixed in Mill `0.11.0-M2`. We already identified the colliding `toolsClasspath` and fixed it partially, but we did not remove it in Mill `0.10.x` for binary compatibility reasons. (https://github.com/com-lihaoyi/mill/issues/916)

The other is probably an issue in the trait inheritance hierarchy, which might be solved by #2274 .

**Comment by lefou** [comment]
* Fixed by #2274
